### DISCUSSION TITLE
Search keywords not highlighted in the search result when searching in a library

### DISCUSSION THREAD

**Discussion by Davidtradeeconomics** [microsoft-search/pnp-modern-search#2109] [Q&A]
Hello all,
I have created a search page using modern PnP Modern Search and want to have the keywords entered in the search box to be highlighted in the search result. Initially, when the search was done in the whole SharePoint this capability was available out of the box, but as I wanted the search to be done in a specific library I have added a path in my query template to point to the library I want to search in. after doing that the keywords were not highlighted anymore. I have tried to add "{searchBoxQuery}" or "{searchTerms}" in **Hit-highlighted properties** option but it doesn't work.
Does anyone have faced this and know if it is possible or has an idea how I can make this work?

Thank you

**Comment by wobba** [comment]
You should only enter managed properties in the hithighlighted properties settings if you want highlights to be built from a specific property. Without this you should get hithighlighting back. Do you get highlights when using OOB site search with the same query on a site? Could be an API issue as the web parts will render what is returned.

**Reply by Davidtradeeconomics** [reply]
Thank you for your reply.
Yes, I do get highlights when using the OOB site search.
I have been able to find a way around it and keep getting the highlights by putting the path to the library in the Refinement filters instead of the Query template.
Also, when creating another search page with another set of documents, I noticed that the search keywords are not highlighted in some documents in the OOB site search and therefore in the PnP modern search as well. Any idea as to why is that and if we can modify this to make the keywords be highlighted? the files are in PDF format.

**Reply by wobba** [reply]
Hithighlighting from the API is a black box, and should "just" work. But sometimes it doesn't for some reason, and not sure when/why.
